docs/source/core_services/drivers/driver_configuration/modbus-tk-driver.rst

```diff
@@ -0,0 +1,263 @@
+but only **device_address** is required:
+
+    - **name** (Optional) - Name of the device. Defaults to "UNKNOWN".
+    - **device_type** (Optional) - Name of the device type. Defaults to "UNKNOWN".
+    - **device_address** (Required) - IP Address of the device.
+    - **port** (Optional) - Port the device is listening on. Defaults to 0 (no port). Use port 0 for RTU transport.
+    - **slave_id** (Optional) - Slave ID of the device. Defaults to 1. Use ID 0 for no slave.
+    - **baudrate** (Optional) - Serial (RTU) baud rate. Defaults to 9600.
+    - **bytesize** (Optional) - Serial (RTU) byte size: 5, 6, 7, or 8. Defaults to 8.
+    - **parity** (Optional) - Serial (RTU) parity: none, even, odd, mark, or space. Defaults to none.
+    - **stopbits** (Optional) - Serial (RTU) stop bits: 1, 1.5, or 2. Defaults to 1.
+    - **xonxoff** (Optional) - Serial (RTU) flow control: 0 or 1. Defaults to 0.
+    - **addressing** (Optional) - Data address table: offset, offset_plus, or address. Defaults to offset.
+        - address: The exact value of the address without any offset value.
+        - offset: The value of the address plus the offset value.
+        - offset_plus: The value of the address plus the offset value plus one.
+        - : If an offset value is to be added, it is determined based on a point's properties in the CSV file:
+            - Type=bool, Writable=TRUE:       0
+            - Type=bool, Writable=FALSE:  10000
+            - Type!=bool, Writable=TRUE:  30000
+            - Type!=bool, Writable=FALSE: 40000
+    - **endian** (Optional) - Byte order: big, little, or mixed. Defaults to big.
+    - **write_multiple_registers** (Optional) - Write multiple coils or registers at a time. Defaults to true.
+        - : If write_multiple_registers is set to false, only register types unsigned short (uint16) and boolean (bool)
+        is supported. The exception raised during the configure process.
+    - **selected_registers** (Optional) - Comma-separated list of "Register Name" values, selected from the
+      registry_config. Defaults to all.
+        - example: "register_a,register_b,register_c"
+
+Sample Modbus-TK configuration files are checked into the VOLTTRON repository
+in ``services/core/MasterDriverAgent/master_driver/interfaces/modbus_tk/maps``.
+
+Here is a sample TCP/IP Modbus-TK device configuration:
+
+.. code-block:: json
+
+    {
+        "driver_config": {
+            "device_address": "10.1.1.2",
+            "port": "5020"
+        },
+        "driver_type": "modbus_tk",
+        "registry_config":"config://modbus_tk_test.csv",
+        "interval": 60,
+        "timezone": "UTC",
+        "heart_beat_point": "heartbeat"
+    }
+
+Here is a sample RTU Modbus-TK device configuration, using all default settings:
+
+.. code-block:: json
+
+    {
+        "driver_config": {
+            "device_address": "10.1.1.2"
```

Comment:
Should probably change this to "/dev/tty.usbserial-AL00IEEY" is this is RTU.

Reply:
Thanks for your comment. I'll change it to "/dev/tty.usbserial-AL00IEEY".